execution-core/src/tx.rs

```diff
@@ -0,0 +1,259 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Types used by the wallets and the circuit to represent transations
+//! before proving. This will be used by the wallets and other users
+//! who wish to sign transations offline
+
+use alloc::string::String;
+use alloc::vec::Vec;
+
+use dusk_bytes::{DeserializableSlice, Error as BytesError};
+use jubjub_schnorr::SignatureDouble;
+use piecrust_uplink::{ContractId, CONTRACT_ID_BYTES};
```

Comment:
To avoid depending on piecrust-uplink here, we define our own `ContractId` in the `transfer` module